src/lib/button-toggle/button-toggle.ts

```diff
@@ -75,135 +68,183 @@ export class MdButtonToggleGroup implements AfterViewInit, ControlValueAccessor
+  @Input()
+  get name(): string { return this._name; }
   set name(value: string) {
     this._name = value;
     this._updateButtonToggleNames();
   }
 
   /** Whether the toggle group is disabled. */
   @Input()
-  get disabled(): boolean {
-    return this._disabled;
-  }
-
-  set disabled(value) {
-    this._disabled = coerceBooleanProperty(value);
-  }
+  get disabled(): boolean { return this._disabled; }
+  set disabled(value) { this._disabled = coerceBooleanProperty(value); }
 
   /** Whether the toggle group is vertical. */
   @Input()
-  get vertical(): boolean {
-    return this._vertical;
-  }
+  get vertical(): boolean { return this._vertical; }
+  set vertical(value) { this._vertical = coerceBooleanProperty(value); }
+
+  /** Whether the toggle group is disabled. */
+  @Input()
+  get multiple(): boolean { return this._multiple; }
+  set multiple(value) { this._multiple = coerceBooleanProperty(value); }
 
-  set vertical(value) {
-    this._vertical = coerceBooleanProperty(value);
+  /** The selected value in the input group. */
+  get selected(): MdButtonToggle|MdButtonToggle[] {
+    return this.multiple ? this._model.selected : this._model.selected[0];
   }
 
   /** Value of the toggle group. */
   @Input()
   get value(): any {
-    return this._value;
+    if (this._model && !this._model.isEmpty()) {
+      return Array.isArray(this.selected) ?
+        this.selected.map(toggle => toggle.value) :
+        this.selected.value;
+    }
   }
 
-  set value(newValue: any) {
-    if (this._value != newValue) {
-      this._value = newValue;
-
-      this._updateSelectedButtonToggleFromValue();
+  set value(newValue: any|any[]) {
+    if (this._model) {
+      this._selectByValue(newValue);
+    } else {
+      this._tempValue = newValue;
+    }
+  }
 
-      // Only emit a change event if the view is completely initialized.
-      // We don't want to emit a change event for the initial values.
-      if (this._isInitialized) {
-        this._emitChangeEvent();
-      }
+  /** Syncs the `name` attribute of the child toggles with the `name` of the group. */
+  private _updateButtonToggleNames(): void {
+    if (this._toggles) {
+      this._toggles.forEach(toggle => toggle.name = this._name);
     }
   }
 
-  /** Whether the toggle group is selected. */
-  @Input()
-  get selected() {
-    return this._selected;
+  /** Unsubscribes from the change listener on the child toggles. */
+  private _dropChangeSubscription(): void {
+    if (this._changeSubscription) {
+      this._changeSubscription.unsubscribe();
+      this._changeSubscription = null;
+    }
   }
 
-  set selected(selected: MdButtonToggle) {
-    this._selected = selected;
-    this.value = selected ? selected.value : null;
+  /** Subscribes to user-generated changes of the button toggles. */
+  private _subscribeToToggleChanges(): void {
+    let source = Observable.merge.apply(Observable, this._toggles.map(toggle => toggle.change));
```

Comment:
I haven't really used `mergeMap` before, but it doesn't seem like it's appropriate here. We only want to merge all of the subscriptions into one so it's easier to listen and unsubscribe to all of them.